sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java

```diff
@@ -3806,19 +3822,31 @@ public void drainEncoder(boolean endOfStream) {
         sdlSession.drainEncoder(endOfStream);
     }
 	
-	private void NavServiceStarted() {
+	private void NavServiceStarted(boolean isEncrypted) {
 		navServiceStartResponseReceived = true;
 		navServiceStartResponse = true;
+
+		if(sdlSession.getServiceListeners() != null && sdlSession.getServiceListeners().containsKey(SessionType.NAV)){

```

Comment:
This logic for callbacks should all be in the SdlSession class. Just use the logic in methods like [`onProtocolSessionStarted`](https://github.com/smartdevicelink/sdl_android/blob/d8d92bc44399b5a552fa3211e0d9928ab3d07454/sdl_android/src/main/java/com/smartdevicelink/SdlConnection/SdlSession.java#L488)